oauthenticator/azuread.py

```diff
@@ -19,6 +20,10 @@
 PYJWT_2 = V(jwt.__version__) >= V("2.0")
 
 
+def check_user_has_role(member_roles, allowed_roles):
```

Comment:
Hmmm, this function name:

1. Let's prefix it with `_`
2. Let's reconsider naming of this function and its arguments.
   It receives a user's app roles and checks for any of the allowed roles, but from the name of the function as it is, I would expect a user and a role to be passed.
   So perhaps something more like `_check_if_any_user_role_is_an_allowed_role`?

Perhaps the readability is best improved by letting this one line function be inlined where its used rather than updating its name?

Reply:
@consideRatio I was just trying to create something similar to what I saw in google provider. There you have something more wordy than my one-liner:

```python
def check_user_in_groups(member_groups, allowed_groups):
    # Check if user is a member of any group in the allowed groups
    if any(g in member_groups for g in allowed_groups):
        return True  # user _is_ in group
    else:
        return False
```

I think maybe we should just get rid of `check_user_has_role()` at all.
Since we moved from `List` to `Set` in `AzureAdOAuthenticator`, how do you feel about something like this?

```python
        roles = set(auth_state['user'].get("roles", []))

        if self.admin_users_app_roles:
            if roles.intersection(self.admin_users_app_roles):
                userdict["admin"] = True
                return userdict

        if self.allowed_users_app_roles:
            if not roles.intersection(self.allowed_users_app_roles):
                return None
```

I know that `intersection` can be replaced with `&`, though I rarely get to see sets, so, IMO, `intersection` improves readability. I hope I'm not the only one who prefers it. :)

Reply:
I fully agree on using intersection rather than `&` for readability :D I didn't know about `&` btw :)

Reply:
> @consideRatio I was just trying to create something similar to what I saw in google provider.

Ah yeah, I'm not a fan of that implementation now that I know about it ;D The function name was not obvious, so an inline comment needed to correct for it etc. Sorry for having you need to experience review friction even though similar code have been accepted in the past.

+1 For not having such function at all and just letting the code speak for itself like you suggested above!

Reply:
@consideRatio No worries :) It's great that we now have a better piece of code that can later be set as a good example for someone else. The review process was worth it for sure. Thanks for helping me on this!